src/decision/justification_heuristic.h

```diff
@@ -38,7 +38,15 @@
 namespace CVC4 {
 namespace decision {
 
-class JustificationHeuristic : public ITEDecisionStrategy {
+template <bool ref_count>
+class JustificationHeuristic;
+
+template <bool check2>
+class JustificationHeuristic : public ITEDecisionStrategy
+{
+  friend class JustificationHeuristic<true>;
+  friend class JustificationHeuristic<false>;
```

Comment:
Are those `friend` declarations necessary?

Reply:
You're right. I don't think they are.